MIRest/src/com/cubeiosample/webservices/rest/jersey/MovieRentals.java

```diff
@@ -341,6 +341,190 @@ public JSONObject returnMovie(int inventoryId, int customerId, int staffId, doub
+		}
+		return  rs.getJSONObject(0);
+	}
+
+	public JSONObject updateGenreGroupName(String name , int id) {
+		String query = "Update genre_group SET name=? where genre_group_id = ?";
+		JSONArray params = new JSONArray();
+		RestOverSql.addStringParam(params, name);
+		RestOverSql.addIntegerParam(params, id);
+		return ros.executeUpdate(query, params);
+	}
+
+	public JSONObject createGenreGroup(String name, int customerId) {
+		String query = "INSERT INTO genre_group (name, customer_id) VALUES (?, ?)";
+		JSONArray params = new JSONArray();
+		RestOverSql.addStringParam(params, name);
+		RestOverSql.addIntegerParam(params, customerId);
+		return ros.executeUpdate(query, params);
+	}
+
+	public JSONArray getAllCategories() {
+		String query = "Select category_id, name from category";
+		JSONArray params = new JSONArray();
+		Map<String, Integer>  categoryIdValues = new HashMap<>();
+		return ros.executeQuery(query, params);
+	}
+
+	public JSONArray getCategoriesByNames(List<String> categories) {
+		String query = "Select category_id, name from category where name In (";
+		JSONArray params = new JSONArray();
+		for(int i=0; i< categories.size() && i < 60; i++) {
+			query = query + "?";
+			if (i < categories.size()-1)
+				query = query + ",";
+			if(i == categories.size()-1)
+				query = query + ")";
+			RestOverSql.addStringParam(params, categories.get(i));
+		}
+		return ros.executeQuery(query, params);
+	}
+
+	public JSONArray getCategoriesByIds(List<Integer> categories) {
+		String query = "Select category_id, name from category where category_id In (";
+		JSONArray params = new JSONArray();
+		for(int i=0; i< categories.size() && i < 60; i++) {
+			query = query + "?";
+			if (i < categories.size()-1)
+				query = query + ",";
+			if(i == categories.size()-1)
+				query = query + ")";
+			RestOverSql.addIntegerParam(params, categories.get(i));
+		}
+		return ros.executeQuery(query, params);
+	}
+
+	private Map<String, Integer> categoryNameIdMap(List<String> categories) {
+		Map<String, Integer>  nameIdMap = new HashMap<>();
+		JSONArray rs = getCategoriesByNames(categories);
+		for(Object obj: rs) {
+			JSONObject jsonObject = (JSONObject)obj;
+			nameIdMap.put(jsonObject.getString("name"), jsonObject.getInt("category_id"));
+		}
+		return  nameIdMap;
+	}
+
+	public void deleteGenreGroupCategoryMapping(int genreGroupId) {
+		String genreGroupQuery = "delete from genre_group_category "
+				+ " where genre_group_id = ?";
+		JSONArray params = new JSONArray();
+		RestOverSql.addIntegerParam(params, genreGroupId);
+		ros.executeUpdate(genreGroupQuery, params);
+	}
+
+	public void deleteGenreGroup(int genreGroupId) {
+		String genreGroupQuery = "delete from genre_group "
+				+ " where genre_group_id = ?";
+		JSONArray params = new JSONArray();
+		RestOverSql.addIntegerParam(params, genreGroupId);
+		ros.executeUpdate(genreGroupQuery, params);
+	}
+
+	public boolean genre_group_category_mapping(List<String> categories, int genreGroupId) {
```

Comment:
How do handle the case where have to delete a category from category group

Reply:
For update part we are removing all the old mapping and updating the new ones sent from UI.
e.g If group initially contains Action and Romance, and we removed Romance, when save is called only Action will be sent the old mapping gets deleted and we will save new mapping for Action only